fmn/api/distgit.py

```diff
@@ -0,0 +1,25 @@
+import logging
+
+import requests
+from fastapi import Depends
+
+from ..core.config import Settings, get_settings
+
+log = logging.getLogger(__name__)
+
+
+class DistGitClient:
+    def __init__(self, settings):
+        self.url = settings.services.distgit_url
+        self.session = requests.Session()
+
+    def get_artifacts(self, pattern):
+        url = f"{self.url}/api/0/projects"
+        projects = self.session.get(
+            url=url, params={"pattern": f"*{pattern}*", "short": "true"}
+        ).json()
```

Comment:
Please use `response.raise_for_status()` to get an exception when we can't reach distgit. You'll call `.json()` after that. See `fmn/api/auth.py` for an example.